lib/iris/tests/integration/test_pickle.py

```diff
@@ -31,35 +31,43 @@
     from iris.fileformats.grib.message import GribMessage
 
 
+class Common(object):
+    def pickle_cube(self, path):
+        # Ensure that data proxies are pickleable.
+        cube = iris.load(path)[0]
+        with self.temp_filename('.pkl') as filename:
+            with open(filename, 'wb') as f:
+                pickle.dump(cube, f)
+            with open(filename, 'r') as f:
+                ncube = pickle.load(f)
+        self.assertEqual(ncube, cube)
+
+
 @tests.skip_data
 @tests.skip_grib
-class TestGribMessage(tests.IrisTest):
-    def test(self):
-        # Check that a GribMessage pickles without errors.
-        path = tests.get_data_path(('GRIB', 'fp_units', 'hours.grib2'))
-        messages = GribMessage.messages_from_filename(path)
-        message = next(messages)
+class TestGribMessage(Common, tests.IrisTest):
+    def setUp(self):
+        self.path = tests.get_data_path(('GRIB', 'fp_units', 'hours.grib2'))
+
+    def pickle_obj(self, obj):
         with self.temp_filename('.pkl') as filename:
             with open(filename, 'wb') as f:
-                pickle.dump(message, f)
+                pickle.dump(obj, f)
+
+    def test(self):
+        # Check that a GribMessage pickles without errors.
+        messages = GribMessage.messages_from_filename(self.path)
+        obj = next(messages)
+        self.pickle_obj(obj)
```

Comment:
That's right, these are the original two tests from this module which I refactored only to perform a common pickle dump (self.pickle_obj).  I have added `test_roundtrip` which performs a dump and load and performs an assert.

I do not know enough about grib to understand whether `test` and `test_data` actually test something that my new `test_roundtrip` does not (i.e. redundant or not).  Can you confirm whether these two tests still serve a purpose?